queue/min_time_heap.go

```diff
@@ -0,0 +1,50 @@
+package queue
```

Comment:
Talked about this offline,but just leaving this here for others to give opinion: I'd suggest to use the `queue.PriorityQueue` type intorduced in #3694 as that does not require `container/heap` for usage and more general by allowing arbitrary values which have `Less` defined (and by this not forcing a particular order).

Reply:
`queue.PriorityQueue` cherry picked